src/Microsoft.Azure.SignalR.Protocols/MulticastDataMessage.cs

```diff
@@ -198,4 +198,101 @@ public MultiGroupBroadcastDataMessage(IReadOnlyList<string> groupList, IDictiona
             GroupList = groupList;
         }
     }
-}
+
+    /// <summary>
+    /// A data message to indicate a client invocation request.
+    /// </summary>
+    public class ClientInvocationMessage : MulticastDataMessage
```

Comment:
invocation message is always to that client, why is it multicast?

Reply:
To make it simple and follow send to connection message, where service will pick up the protocol.

Reply:
Looks there's some concern about `ConnectionDataMessage` is used for handshake. See [comment](https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManager.cs#L110-L112).
Besides, for remote connection, server may not have the knowledge about client connection protocol.